### ISSUE TITLE
Omnibus Chef `gem install` fails when trying to compile C extensions

### ISSUE THREAD

**Issue by bixu** [adamhjk/dynect_rest#41]
I don't understand why gem-install here is trying to build a C extension...
```
ubuntu@2f1b3bc4-4fa3-4e98-8e15-a9cde5e9bd43:~$ /opt/chef/embedded/bin/ruby --version
ruby 1.9.3p547 (2014-05-14 revision 45962) [x86_64-linux]

ubuntu@2f1b3bc4-4fa3-4e98-8e15-a9cde5e9bd43:~$ /opt/chef/bin/chef-client --version
Chef: 11.16.4

ubuntu@2f1b3bc4-4fa3-4e98-8e15-a9cde5e9bd43:~$ cat /etc/product
Name: Joyent Instance
Image: Ubuntu 14.04.2 LTS 20150225.2
Documentation: https://docs.joyent.com/images/linux/ubuntu-certified

ubuntu@2f1b3bc4-4fa3-4e98-8e15-a9cde5e9bd43:~$ uname -a
Linux 2f1b3bc4-4fa3-4e98-8e15-a9cde5e9bd43 3.13.0-46-generic #75-Ubuntu SMP Tue Feb 10 15:24:04 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

ubuntu@2f1b3bc4-4fa3-4e98-8e15-a9cde5e9bd43:~$ sudo /opt/chef/embedded/bin/gem install dynect_rest -q --no-rdoc --no-ri -v "0.4.3"
Building native extensions.  This could take a while...
ERROR:  Error installing dynect_rest:
	ERROR: Failed to build gem native extension.

        /opt/chef/embedded/bin/ruby extconf.rb
checking for main() in -lstdc++... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of
necessary libraries and/or headers.  Check the mkmf.log file for more
details.  You may need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/opt/chef/embedded/bin/ruby
	--with-stdc++lib
	--without-stdc++lib
/opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:381:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:461:in `try_link0'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:476:in `try_link'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:619:in `try_func'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:845:in `block in have_library'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:790:in `block in checking_for'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:284:in `block (2 levels) in postpone'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:254:in `open'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:284:in `block in postpone'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:254:in `open'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:280:in `postpone'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:789:in `checking_for'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:840:in `have_library'
	from extconf.rb:2:in `<main>'


Gem files will remain installed in /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/unf_ext-0.0.6 for inspection.
Results logged to /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/unf_ext-0.0.6/ext/unf_ext/gem_make.out
```

**Comment by bixu** [comment]
@adamhjk, it looks like the `json` gem now requires building native extensions - we could instead depend on `json_pure`, which would keep things a lot lighter/more portable: https://rubygems.org/gems/json_pure

**Comment by ranjib** [comment]
probably json binding were missing during ruby comilation/build cause json is part of the stdlib from 1.9 onwards, http://ruby-doc.org/stdlib-1.9.3/libdoc/json/rdoc/JSON.html.
I'll drop the json gem requirement and test things out.

@bixu sorry for getting back late, somehow i was not watching the repo, hence i didnt get notified when you filed the issue. Fixed that.

We are likely not to include json_pure or any other dependency if stdlib has a counterpart.

**Comment by bixu** [comment]
Thanks!  I thought about forking the gem but realized that was probably counterproductive if we could just fix in upstream.  I fully support stdlib instead of a dep if we can do that :)

**Comment by bixu** [comment]
@ranjib Any ETA?  This is currently making my Test Kitchen life hell :)

**Comment by ranjib** [comment]
@bixu i have removed json gem dependency and released 0.4.4 version of the gem. Let me know if your problem still persist

**Comment by bixu** [comment]
Looks like it's failing still on the `und` dep:
```
root@a8d8b9df-d377-c037-c61e-b176a36afabd:~# /opt/chef/embedded/bin/gem install unf
Building native extensions.  This could take a while...
ERROR:  Error installing unf:
	ERROR: Failed to build gem native extension.

        /opt/chef/embedded/bin/ruby extconf.rb
checking for main() in -lstdc++... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of
necessary libraries and/or headers.  Check the mkmf.log file for more
details.  You may need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/opt/chef/embedded/bin/ruby
	--with-stdc++lib
	--without-stdc++lib
/opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:381:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:461:in `try_link0'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:476:in `try_link'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:619:in `try_func'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:845:in `block in have_library'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:790:in `block in checking_for'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:284:in `block (2 levels) in postpone'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:254:in `open'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:284:in `block in postpone'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:254:in `open'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:280:in `postpone'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:789:in `checking_for'
	from /opt/chef/embedded/lib/ruby/1.9.1/mkmf.rb:840:in `have_library'
	from extconf.rb:2:in `<main>'


Gem files will remain installed in /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/unf_ext-0.0.6 for inspection.
Results logged to /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/unf_ext-0.0.6/ext/unf_ext/gem_make.out
```
